modo/io.py

```diff
@@ -59,19 +59,33 @@ def parse_multiple_instances(path: Path) -> List:
 def build_modo_from_file(path: Path, object_directory: Path) -> MODO:
     """build a modo from a yaml or json file"""
     instances = parse_multiple_instances(Path(path))
+    # check for unique ids and fail early
+    ids = [inst.id for inst in instances]
+    if len(ids) > len(set(ids)):
+        dup = {x for x in ids if ids.count(x) > 1}
+        raise ValueError(
+            f"Please specify a unique ID. Element(s) with ID(s) {dup} already exist."
+        )
     modo_inst = [
         instance for instance in instances if isinstance(instance, model.MODO)
     ]
     if len(modo_inst) != 1:
         raise ValueError(
             f"There must be exactly 1 MODO in the input file. Found {len(modo_inst)}"
         )
-    # TODO: Could we rename id_ to id?
     modo_dict = modo_inst[0]._as_dict
     modo = MODO(path=object_directory, **modo_dict)
     for instance in instances:
         if not isinstance(instance, model.MODO):
-            modo.add_element(instance)
+            if (
+                isinstance(instance, model.DataEntity)
+                and not modo.path in Path(instance.data_path).parents
```

Comment:
yes, I'm sure, but I think you are right and currently that is not a real use case, as the modo will usually be created with `build_modo_from_file()` and thus the path should not exist within modo. I thought this could be useful if we also allow updating modo's via yaml file, but currently that is anyway not implemented/working. Happy to remove the check?